test/e2e/storage/testsuites/volume_io.go

```diff
@@ -74,87 +74,59 @@ func (t *volumeIOTestSuite) getTestSuiteInfo() TestSuiteInfo {
 	return t.tsInfo
 }
 
-func (t *volumeIOTestSuite) skipUnsupportedTest(pattern testpatterns.TestPattern, driver TestDriver) {
-}
-
-func createVolumeIOTestInput(pattern testpatterns.TestPattern, resource genericVolumeTestResource) volumeIOTestInput {
-	var fsGroup *int64
-	driver := resource.driver
-	dInfo := driver.GetDriverInfo()
-	f := dInfo.Config.Framework
-	fileSizes := createFileSizes(dInfo.MaxFileSize)
-	volSource := resource.volSource
-
-	if volSource == nil {
-		framework.Skipf("Driver %q does not define volumeSource - skipping", dInfo.Name)
+func (t *volumeIOTestSuite) defineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	var (
+		dInfo       = driver.GetDriverInfo()
+		config      *PerTestConfig
+		testCleanup func()
+		resource    *genericVolumeTestResource
+	)
+
+	// No preconditions to test. Normally they would be in a BeforeEach here.
+
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("volumeio")
+
+	init := func() {
+		// Now do the more expensive test initialization.
+		config, testCleanup = driver.PrepareTest(f)
+		resource = createGenericVolumeTestResource(driver, config, pattern)
+		if resource.volSource == nil {
```

Comment:
nit, check not needed

Reply:
Probably not, but it was there in the original code and I wanted to avoid changing test logic.